src/main/kotlin/com/hj/leetcode/kotlin/problem705/Solution.kt

```diff
@@ -0,0 +1,62 @@
+package com.hj.leetcode.kotlin.problem705
+
+/**
+ * LeetCode page: [705. Design HashSet](https://leetcode.com/problems/design-hashset/);
+ */
+class MyHashSet() {
+
+    private val initialTableSize = 16
+    private var table = Array(initialTableSize) { mutableListOf<Int>() }
+    private var numElements = 0
+
+    fun add(key: Int) {
+        val slot = hashSlot(key)
+        if (slot.contains(key)) {
+            return
+        }
+
+        slot.add(key)
+        numElements++
+        migrateToLargerTableAsNeeded()
+    }
+
+    private fun hashSlot(key: Int): MutableList<Int> {
+        return table[hashCode(key)]
+    }
+
+    private fun hashCode(key: Int): Int {
+        return Math.floorMod(key, table.size)
+    }
+
+    private fun migrateToLargerTableAsNeeded() {
+        if (shouldMigrateToLargerTable()) {
+            migrateToLargerTable()
+        }
+    }
+
+    private fun shouldMigrateToLargerTable(): Boolean {
+        return numElements / table.size > 0.75
+    }
+
+    private fun migrateToLargerTable() {
+        val oldTable = table
+        table = Array(oldTable.size * 2) { mutableListOf<Int>() }
+
+        for (slot in oldTable) {
+            for (element in slot) {
+                hashSlot(element).add(element)
+            }
+        }
+    }
+
+    fun remove(key: Int) {
```

Comment:
Missing judgement on the time and space complexity

Reply:
Has marked a TODO for the complexity analysis at https://github.com/hj-core/hj-leetcode-kotlin/pull/72/commits/19e0967aeb3b55b254331a2abe9a6406f83916c5